### ISSUE TITLE
ts-brand no longer passing tsc 3.9.7 and higher

### ISSUE THREAD

**Issue by benallfree** [kourge/ts-brand#2]
```bash
$ git clone git clone git@github.com:kourge/ts-brand.git
$ cd ts-brand
$ tsc --version
Version 4.2.4

$ tsc
src/index.ts:44:42 - error TS2536: Type '"__witness__"' cannot be used to index type 'B'.

44 export type BaseOf<B extends AnyBrand> = B['__witness__'];
                                            ~~~~~~~~~~~~~~~
```

Or paste src/index.ts into the [typescript playground](https://www.typescriptlang.org/play?noStrictGenericChecks=true&ts=3.9.7#code/PQKhCgAIUhBSAGAhATgQwHYBMGQJYDOkakALgJ4AOApmQBZqlloDW1RjkAtnhnlwFcuZAO4B7MlVqU06LtVLUUBAHSQA4ngBu1DMUgAjNAWpQYFGoiTHquTFkgEx8yAL4BHAbXvEUBvKToKOSG6Ni8AOaSlshhWJEIADT4pGaQlChiWAIAxuz6FrRGJg5iegYCTPEEpLw5TABmmcLINghqACp0tCjsAgA2tRhRBnHUWGmF+EQYYlV4vfX9IcYEeBEYaAb9tE3O9EU20dTJPrNMgqQCaP3Laavrm9u0IgF0xOfdKKH248eQWCU2j+e2EpG6jjQLmKtEKKjMaQ6rFoOVksIYTHBRTikX+hFcHi8nXEP2wf0KRHBnAIDF6BzSBChhxM-x8o3suKmaMgOTKa0BvQcWMhLkKAEJIF0lN46aRxGkRGhyJSJGgtGI8EK6IR4dBERCGgsapBFSF8aQJAQBJRKMsCtqUEKpOlZEzFChkggAErsJQ6LAAOSZSRNb2IDMCuP6ASUN3+VKY+POiAA+inCmmQyJtTl3uaIYCGmgBqRdRBoJKISZedgTUrpmQJIXeN5SYCnZZeGQlFwiGIGikiFaUJk3PFhmleO6i3lkljegYQg0xCgeMN9IsBMpgcdOtqiPjKGICGtnoZqKiBCz56YK9zZjy+YE0FO+3ps5xhezwuupoJjQY1BqAAklUYj5MmDA6I2aSAfoNQoFGMboP0xzJNQOh6OCYgCBEeZMP2ih6DqCIVgAAtQAAeUK2reMAIAxkzOgACow7wALyQKg9gADwIZEyQAOQyOCgkAHwANxMZYACqJgoMBDicdx2A8RgQiAR6kCCVeSjiVJFZTAAIngDQNNKGCkHJSiKZAylxGpGlKEJukoIJQlpiwvBYGm+mTpZSgzrQzHHkwADe+BYAAXFxDnqVwmnJCFNRiZAAC+aQMQgZjAOAVFHigmLOipWA8VAXE2Ik5UlQJ5U+vJ-pBi4VFEVgQ6RuunGCWmGYpoJ4CpcpRwAGSQGFADaADS+B6PVfrjE11AALoxTVwxpZAo1hWmrykBg7AEGmq02GlUngKA5YwLAegINd5Albg+IkN+7b-DCpR6JgIQwm9YwOLt7xfbBOK-lIahIGapAmP0A4BA2ya6Q0AzJBUiZMKin31MQlLOjWCEvpZCrdHoXCYHglADIwuIveSUgHnoES6LG-Rlrl+UrkVlh3SVdmxbxX2nBg5CSeA51gJl1gmAA8g0j0cPGGLMGwHBC22tMxEgdi1uQeDUP0bWDoYRxwjleWUQVnO0JL1AyzxSCQC1ugG9zcSDVx43dSmu37SevlLWdF1pPAsS-Cgcv6EjGD1HgZT0J+yLy1oNxeJA-aOM4zKws6PiojUBCTIrSf9CnFr6DTHa0AKwIOKCkKakb15g5AoE8pg570DheGpBWafGAc-y95AABEeOBLkFooEPyRdsKkfR2UcYZGIERyGuUQmBgLL9gXLwrlgZZpBR1FcLRmWMYZzrWQpSl86p8WJdprl+RWwDAHAt8CjxV+Kal4gDEKyJ9D3yUB8BwvQrgoE+pAb+EwKxZVNuzQq-wSpKDtg7SirUiAu3sG7AAFGOJQyxIjHWlg0O2YkACUdlUpIADuLCs8BGb7UQjkSAc9aixwTMkbMuhIARGBHoJw0I-poVblhQBJAi4pwBjvBuWdLBpy-CIwogsHC50xIXZOsIJBKN+BXMszdGgDENLcSkEIazPixoPEOZIw4HwYUwFAbhajyDnPuNhbh56CP4LaEI4Ctyb37lI7wBAAC0JE9TkRkHIAkAoiHrilJAYJoZwRl2NlIOcEg4LqPGIffxkDKy0EZC4AhKB4lRGCSjSorcai5PIlRGiOwz7ZQvrJeStl7K8WAVpHS8ln4wFHtA9p0UP6oJgW7TUuhagUAMvRc+IA2bmw5h4qOHDiKAksgEcgaDHbYCwULEqYl8G2PKSQm2ZCkCUNWmNcq+S9ClPKdjLiUkMpi0upAZimRsh5HljYz+lyQz8MwmXOI-xkDtErPiNYJ87R3PznAyZmyKC4FRpAXo0Z8icOdLwcyvQo7aL4TuYUdyFYohXIsbu5gwakRgGRElCBEXTPIC02lDSYV0UQPM-4MDeYlUcglZyj8+kiwrIMnlnFSZsC-sMo5FDZmPk3hccg4qhk2SwLggALAAJjlc0hBSykHsJjiTZEOyMFO32fdV2uCKGrTGCgchNzIBooUAEyKUytkvPAEAA)

**Comment by yf-yang** [comment]
AnyBrand is inferred as any, other attributes are directly discarded.
To solve the problem, change the first argument of AnyBrand.
``` typescript
// Language service says Newtype = any
type NewType = any & {someKey: 'someValue'};

// Try this:
type AnyBrand = Brand<unknown, any>;
```

**Comment by kourge** [comment]
Fixed with the merge of #4.
